tests/datatypes/test_dict.py

```diff
@@ -229,6 +229,66 @@ def test_method_get(self):
                 print(err)
             """)
 
+    def test_copy(self):
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(x == y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            print(x is not y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {42: 'Babel'}
+            y = x.copy()
+            y['42'] = 'Babel'
+            print(x == y)
+            """)
+
+        self.assertCodeExecution("""
+            x = {'key': {42: 'Babel'}, 42: 'Babel'}
+            y = x.copy()
+            print(x['key'] is y['key'])
+            """)
+
+    def test_fromkeys(self):
+        self.assertCodeExecution("""
+            keys = [1, 2]
+            print({}.fromkeys(keys))
+            """)
+
+        # non-iterable error test
+        self.assertCodeExecution("""
+            keys = [1, 2]
+            value = 2
+            print({}.fromkeys(keys, value))
+            """)
+
+        self.assertCodeExecution("""
+            keys = (1, 2)
+            value = 5
+            print({}.fromkeys(keys, 5))
```

Comment:
It looks like `value` is not being used here :) 

That said, I think this and the above two cases can be combined into one `self.assertCodeExecution()` – this current approach would work, and we do want the tests to be correct more than we want them to be fast, but each `self.assertCodeExecution()` call takes time to run so we prefer to reduce those where possible.

(I realise that some of the existing tests similarly have multiple `self.assertCodeExecution()` that aren't fully necessary, but we're working on gradually combining those where appropriate!)